records/generator.go

```diff
@@ -176,12 +176,12 @@ func (rg *RecordGenerator) loadFromMaster(ip string, port string) (sj StateJSON)
 	if err != nil {
 		logging.Error.Println(err)
 	}
-	defer resp.Body.Close()
 
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
 		logging.Error.Println(err)
 	}
+	_ = resp.Body.Close()
```

Comment:
Yes, and we should totally use defer whenever we can use it. It's my fault I didn't explain this change well enough. So, here's the deal: One of the linters in our CI is `errcheck` which detects ignored errors in Go programs. Unfortunately, it creates tension with the common `defer r.Close()` Go idiom. Now, errors returned by `Close` methods can be safely ignored *as long as* what was closed was read-only. Whenever there are writes involved, errors returned by close should absolutely not be ignored.

What is lacking, then, is the appropriate tooling to handles such intricacies intelligently. In this PR, I did the easiest thing that would silence the CI failure, but I knew that it wasn't the right thing to do. I will follow up in https://github.com/mesosphere/mesos-dns/issues/223